tests/integration/master.test.js

```diff
@@ -0,0 +1,255 @@
+    let key = u8a_hex(randomish_u8a());
+    let val = u8a_hex(randomish_u8a());
+    const did_to_key = [
+      [BOB_DID, await keypair(BOB_SK)],
+      [ALICE_DID, await keypair(ALICE_SK)],
+    ];
+    await master_set_storage(nc, key, val, did_to_key);
+    let sto = await nc.rpc.state.getStorage(key);
+    let u8a = sto.unwrap();
+    expect(u8a_hex(u8a)).toEqual(val);
+  }, 20000);
+});
+
+// connect to running node
+async function connect() {
+  const extraTypes = {
+    Address: 'AccountId',
+    LookupSource: 'AccountId',
+  };
+  return await ApiPromise.create({
+    provider: new WsProvider(FullNodeEndpoint),
+    types: {
+      ...types,
+      ...extraTypes,
+    },
+  });
+}
+
+// load a DID kp from secret
+async function keypair(seed) {
+  assert(seed instanceof Uint8Array, "wrong type");
+  await cryptoWaitReady();
+  let keyring = new Keyring({ type: 'sr25519' });
+  let key = keyring.addFromSeed(seed);
+  return key
+}
+
+async function sign_send_tx(extrinsic) {
+  let key = await get_test_account_key();
+  await extrinsic.signAsync(key);
+
+  const promise = new Promise((resolve, reject) => {
+    try {
+      let unsubFunc = null;
+      return extrinsic.send(({ events = [], status }) => {
+        if (status.isFinalized) {
+          unsubFunc();
+          resolve({
+            events,
+            status,
+          });
+        }
+      })
+        .catch((error) => {
+          reject(error);
+        })
+        .then((unsub) => {
+          unsubFunc = unsub;
+        });
+    } catch (error) {
+      reject(error);
+    }
+
+    return this;
+  });
+
+  return await promise;
+}
+
+async function get_test_account_key() {
+  await cryptoWaitReady();
+  let keyring = new Keyring(TestKeyringOpts);
+  let key = keyring.addFromUri(TestAccountURI);
+  return key
+}
+
+// represent a Uint8Array as hex with a "0x" prefix
+function u8a_hex(bs) {
+  assert(bs instanceof Uint8Array, "wrong type");
+  return '0x' + [...bs].map(bt => ('0' + bt.toString(16)).slice(-2)).join('');
+}
+expect(u8a_hex(new Uint8Array([]))).toEqual('0x');
+expect(u8a_hex(new Uint8Array([0x01]))).toEqual('0x01');
+expect(u8a_hex(new Uint8Array([0xaa, 0xbb]))).toEqual('0xaabb');
+
+function randomish_u8a() {
+  let ret = new Uint8Array(32);
+  for (let i = 0; i < ret.length; i++) {
+    ret[i] = Math.random() * (2 ** 8);
+  }
+  return ret;
+}
+
+async function master_set_storage(
+  nc, // node client
+  key, // hex encoded bytes
+  val, // hex encoded bytes
+  did_to_key, // list of (did, key) pairs with which to vote
+) {
+  let call = nc.tx.system.setStorage([[key, val]]); // this is a root-only extrinsic
+  let payload = {
+    proposal: [...nc.createType('Call', call).toU8a()],
+    round_no: await nc.query.master.round(),
+  };
+  let encoded_state_change = nc.createType('StateChange', { MasterVote: payload }).toU8a();
+
+  let dtk_sorted = [...did_to_key];
+  dtk_sorted.sort(([dida, _a], [didb, _b]) => compare_u8a_32(dida, didb));
```

Comment:
If you don't care about performance and want to reduce code, taking did as hex and calling `sort` should also work, avoiding the need for `compare_u8a_32`

Reply:
done: 974bafd